python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -268,7 +276,7 @@ def transpile(self,
                 outputs={},
                 attrs={
                     "endpoints": pserver_endpoints,
-                    "sync_mode": self.sync_mode,
+                    "sync_mode": True,
```

Comment:
Maybe add comments about how to determine op's sync mode, for send_barrier, recv ops

Reply:
After discuss with @Yancey1989 , send_barrier should always work in sync_mode, code updated.